client/packages/programs/src/api/hooks/document/useEncounterRegistriesByPrograms.ts

```diff
@@ -22,16 +22,19 @@ export const useEncounterRegistriesByPrograms = (
     api.get
       .documentRegistries({
         filter: {
-          parentId: {
-            equalAny: programIds,
+          type: {
+            equalTo: DocumentRegistryTypeNode.Encounter,
+          },
+          contextId: {
+            equalAny: programs.map(it => it.contextId),
```

Comment:
That would be quite a bit of extra work though (backend changes, backend tests, graphql FE definition and hooks). It would also mean if frontend requirements are changing slightly backend changes are needed as well.